.appveyor.yml

```diff
@@ -17,8 +17,8 @@ install:
   - set PATH=%PYTHON%;%PYTHON%\\Scripts;%PATH%
   # The package installation via our setup.py file uses easy_install, which fails to correctly install lxml.
   # As some Python environments don't have lxml preinstalled, install it here to avoid errors during the execution in those cases.
-  - python -m pip install lxml
+  - python -m pip install lxml nose
```

Comment:
Should `nose` be installed here like it is being done in the travis pipeline?

```
pip install nose --no-binary :all:
```

Reply:
This would be necessary for Python > 3.10.

Why is a manual installation of `nose` here in this PR? Shouldn't this be handled by the `pip install .`?

Reply:
A bit of debugging and it works!

The issue -
- `nose` was installed as a `setup_requires` dependency, which means that whenever a `python setup.py <something>` was executed, `nose` and `lxml` were installed automatically.
- This however does not happen when one installs the package using `pip install .`. In this case, only the true requirements, that is, `PyYAML` is installed.

> In previous versions of setuptools, this used to be accomplished with the setup_requires keyword but is now considered deprecated in favor of the [PEP 517](https://peps.python.org/pep-0517/) style described above. To peek into how this legacy keyword is used, consult our [guide on deprecated practice (WIP)](https://setuptools.pypa.io/en/latest/deprecated/index.html).

Solution -
- I have moved `lxml` and `nose` to a new optional dependency group named `"dev"`. This group can now be installed as - `pip install ".[dev]"`.

Reply:
Thanks! This makes sense and I like the solution. It is certainly much clear to keep test dependencies out of `setup_requires`.

Would you mind updating the [docs](https://github.com/sosy-lab/benchexec/blob/main/doc/DEVELOPMENT.md?plain=1#L33) as well?